### ISSUE TITLE
isatty stdin not working

### ISSUE THREAD

**Issue by bpinto** [fish-shell/fish-shell#1443]
`➜ isatty --help`

>   Description
>
>   isatty tests if a file or file descriptor is a tty. The argument may be
>   in the form of a file path, device, or file descriptor number. Without
>   an argument, standard input is implied.

With *stdin* as argument:
````
➜ .oh-my-fish git:(Paulloz-plugin/sprunge) ✗ echo 'foo' | if isatty stdin
                                                     echo 'if'
                                             else
                                                     echo 'else'
                                             end
if
````

Without an argument:
````
➜ .oh-my-fish git:(Paulloz-plugin/sprunge) ✗ echo 'foo' | if isatty
                                                     echo 'if'
                                             else
                                                     echo 'else'
                                             end
else
````

Shouldn't both have the same behaviour? Installed fish from Git (today):
* 1ffa4d9 - (HEAD, origin/master, origin/HEAD, master) openssl: add instructions for using custom certs (6 days ago) <Jack Nagel>

**Comment by geoff-codes** [comment]
> I think the fix in #1061 is wrong.

Yes, it is. Per [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html):

> ##### The if Conditional Construct

> The if command shall execute a compound-list and use its exit status to determine whether to execute another compound-list.

> The format for the if construct is as follows:

> ```
if compound-list
then
    compound-list
[elif compound-list
then
    compound-list] ...
[else
    compound-list]
fi
```

> The if compound-list shall be executed; if its exit status is zero, the then compound-list shall be executed and the command shall complete. Otherwise, each elif compound-list shall be executed, in turn, and if its exit status is zero, the then compound-list shall be executed and the command shall complete. Otherwise, the else compound-list shall be executed.

> ###### Exit Status

> The exit status of the if command shall be the exit status of the then or else compound-list that was executed, or zero, if none was executed.

So its not that it _eats_ the return status. It returns 0 if the the `if`/`else if` conditional(s) are non-zero, otherwise, it takes the return value of the commands executed.

**Comment by zanchey** [comment]
I think we should try and fix this for 2.2.
